lib/puppet/util/windows/win32/process.rb

```diff
@@ -0,0 +1,428 @@
+    #
+    # Possible +int_priority+ values are:
+    #
+    # * Process::NORMAL_PRIORITY_CLASS
+    # * Process::IDLE_PRIORITY_CLASS
+    # * Process::HIGH_PRIORITY_CLASS
+    # * Process::REALTIME_PRIORITY_CLASS
+    # * Process::BELOW_NORMAL_PRIORITY_CLASS
+    # * Process::ABOVE_NORMAL_PRIORITY_CLASS
+
+    def setpriority(kind, int, int_priority)
+      raise TypeError unless kind.is_a?(Integer)
+      raise TypeError unless int.is_a?(Integer)
+      raise TypeError unless int_priority.is_a?(Integer)
+
+      int = Process.pid if int == 0
+      handle = FFI::WIN32::OpenProcess(PROCESS_SET_INFORMATION, 0 , int)
+
+      if handle == 0
+        raise SystemCallError, FFI.errno, "OpenProcess"
+      end
+
+      begin
+        result = FFI::WIN32.SetPriorityClass(handle, int_priority)
+        raise SystemCallError, FFI.errno, "SetPriorityClass" unless result
+      ensure
+        FFI::WIN32.CloseHandle(handle)
+      end
+
+      return 0
+    end
+
+    private
+
+    def initialize_defaults
+      @hash = {
+        app_name: nil,
+        creation_flags: 0,
+        close_handles: true
+      }
+      @si_hash = nil
+      @procinfo = nil
+    end
+
+    def validate_args(args)
+      raise TypeError, 'hash keyword arguments expected' unless args.is_a?(Hash)
+    end
+
+    def validate_keys(args)
+      args.each do |key, val|
+        key = key.to_s.to_sym
+        raise ArgumentError, "invalid key '#{key}'" unless VALID_KEYS.include?(key)
+
+        hash[key] = val
+      end
+    end
+
+    def validate_startup_info
+      hash[:startup_info].each do |key, val|
+        key = key.to_s.to_sym
+        raise ArgumentError, "invalid startup_info key '#{key}'" unless VALID_SI_KEYS.include?(key)
+
+        si_hash[key] = val
+      end
+    end
+
+    def validate_command_line
+      raise ArgumentError, 'command_line or app_name must be specified' unless hash[:app_name] || hash[:command_line]
+    end
+
+    def procinfo
+      @procinfo ||= FFI::WIN32::PROCESS_INFORMATION.new
+    end
+
+    def hash
+      @hash ||= {}
+    end
+
+    def si_hash
+      @si_hash ||= {}
+    end
+
+    def app
+      wide_string(hash[:app_name])
+    end
+
+    def cmd
+      wide_string(hash[:command_line])
+    end
+
+    def cwd
+      wide_string(hash[:cwd])
+    end
+
+    def password
+      wide_string(hash[:password])
+    end
+
+    def logon
+      wide_string(hash[:with_logon])
+    end
+
+    def domain
+      wide_string(hash[:domain])
+    end
+
+    def env
+      env = hash[:environment]
+      return unless env
+
+      env = env.split(File::PATH_SEPARATOR) unless env.respond_to?(:join)
+      env = env.map { |e| e + 0.chr }.join('') + 0.chr
+      env = wide_string(env) if hash[:with_logon]
+      env
+    end
+
+    def process_security
+      return unless hash[:process_inherit]
+
+      process_security = FFI::WIN32::SECURITY_ATTRIBUTES.new
+      process_security[:nLength] = 12
```

Comment:
I think you can do `FFI::WIN32::SECURITY_ATTRIBUTES.size` instead of hardcoding 12